src/api/getSwapQuote.ts

```diff
@@ -16,6 +17,11 @@ import { getAPIParamsForToken } from './utils/getAPIParamsForToken';
 export async function getSwapQuote(
   params: GetSwapQuoteParams,
```

Comment:
Two notes here:
1. show case all params
2. never pass `lifeCycleStatus` inside APIs. API have to stay atomic, and it should be simple input and simple output. Lfecycle is bounded to a component, and the API should be used even without the need to bound something to a component.

Point 2, is a clear point of reflection of System Design interviews or even how to approach the PP2 interview at Coinbase. You want to better polish your thinking on how to design components and APIs, as that is a key skill for any software eng.

Buy this book today https://www.amazon.com/Pragmatic-Programmer-Journeyman-Master/dp/020161622X/ref=sr_1_2?dib=eyJ2IjoiMSJ9.84Xun4zePRzxcYJXHbNXB1wvZkeWhYzsnLiFASxdDssZHty2sHX0n1IsAHY91IBcN34Cmi35fJiQVNMMiFrkKNz3XE4uM_V6mTzQQ-PhrYJA_l1J95Dirj52U0HTs4DKmBxmG0jJPNxAAikRX-jmaXe9b2Ho-w6xebOyzIT4h5N8NhMtGg2mgM20llcF8I5e7GFIt9rwm_T9R1bS_GzZWZ3xiGVss401XLWI_YOUwFI.BheZx0fh6soDsekBonqo9vuAiOIbQS0ibUS2N_HjxfE&dib_tag=se&keywords=The+Pragmatic+Programmer&qid=1725944624&sr=8-2 and read it back to back a couple of times.

You have to polish the way you think about coding so that you can elevate your coding experience to mastery.